src/engine/pieces/pawn.ts

```diff
@@ -1,13 +1,86 @@
-import Piece from './piece';
+import Piece, { PieceType } from './piece';
 import Player from '../player';
 import Board from '../board';
+import Square, { SquareState } from '../square';
+
+export enum PawnMove {
+    ADVANCE_1,
+    ADVANCE_2,
+    TAKE_DIAGONALLY,
+    EN_PASSANT
+}
 
 export default class Pawn extends Piece {
+    public lastMove: PawnMove | null = null;
+
     public constructor(player: Player) {
-        super(player);
+        super(player, PieceType.PAWN);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+        let validMoves = new Array(0);
+        let currentSquare = board.findPiece(this)
+
+        let rowDirection = (this.player == Player.WHITE) ? 1: -1;
+
```

Comment:
```suggestion
        let rowDirection = (this.player === Player.WHITE) ? 1 : -1;
```

we never want to use `==` as it allows type coercion before comparing. This can lead to non-intuitive results and so is generally not recommended in TypeScript 

e.g. `'100' == 100` is true which we would not want 

instead we use `===`